### ISSUE TITLE
New Rule: implement `markdownlint`'s `MD022`

### ISSUE THREAD

**Issue by tooth-is-silver** [eslint-markdown/eslint-markdown#741]
### Rule details

Add a rule equivalent to markdownlint's MD022.

The rule should require headings of any style to have a configurable number of blank lines above and below them. Headings at the beginning or end of a document should not require blank lines outside the document.

Support equivalents of the following MD022 options:

- `lines_above`: Required blank lines above headings. Default: `1`.
- `lines_below`: Required blank lines below headings. Default: `1`.
- `include_front_matter`: Whether to check spacing between front matter and the first heading. Default: `false`.

The blank-line requirements should also support values for individual heading levels. Provide an autofix that inserts missing blank lines.

### What type of rule is this?

Enforces a consistent style or user preference

### Example code

```markdown
<!-- Incorrect -->

# Heading 1
Some text

Some more text
## Heading 2

<!-- Correct -->

# Heading 1

Some text

Some more text

## Heading 2
```

### Prior Art

- markdownlint: [`MD022` - Headings should be surrounded by blank lines](https://github.com/DavidAnson/markdownlint/blob/main/doc/md022.md)

**Comment by lumirlumir** [comment]
Just a note: This rule needs to be implemented in the `consistent-heading-style` rule, following our implementation of `consistent-code-style`.

**Comment by BHyeonKim** [comment]
May I implement this rule?

**Comment by tooth-is-silver** [comment]
Thanks for your interest! I've assigned this issue to @BHyeonKim
